o1vm/src/interpreters/mips/witness.rs

```diff
@@ -80,7 +99,7 @@ pub struct Env<Fp, PreImageOracle: PreImageOracleT> {
     pub registers: Registers<u32>,
     pub registers_write_index: Registers<u64>,
     pub scratch_state_idx: usize,
-    pub scratch_state: [Fp; SCRATCH_SIZE],
+    pub scratch_state: [ToInverseOrNot<Fp>; SCRATCH_SIZE],
```

Comment:
What do you think about using another type of column which cannot be used in further computations, as it is the case for here?
I would approach it with something like `allocate_auxiliary_variable` (or any better name that can include the idea that it is only to satisfy a single constraint, and should not be used for computation) in the `InterpreterEnv` interface and add a new column next to `ScratchState`.

Reply:
I do also think, but I need to dig more into the idea, that we can simply commit to the polynomial whose evaluations are not the inverse. I do not know if it will improve the efficiency though. We can discuss it offline.

Reply:
@dannywillems 
> add a new column next to ScratchState
That is precisely what I suggested in https://github.com/o1-labs/proof-systems/pull/2047

Reply:
Oopsie. I didn't check before :see_no_evil:. I do have a high preference for @querolita solution.